test/spec/api/v4/leaflet/layer-group.spec.js

```diff
@@ -0,0 +1,135 @@
+var L = require('leaflet');
+var carto = require('../../../../../src/api/v4');
+
+describe('src/api/v4/leaflet/layer-group', function () {
+  var client;
+  var layer;
+  var layerGroup;
+  var map;
+  var mapContainer;
+
+  beforeEach(function () {
+    mapContainer = document.createElement('div');
+    mapContainer.id = 'map';
+    document.body.appendChild(mapContainer);
+
+    client = new carto.Client({
+      apiKey: '84fdbd587e4a942510270a48e843b4c1baa11e18',
+      serverUrl: 'https://{user}.carto.com:443',
+      username: 'cartojs-test'
+    });
+    map = L.map('map').setView([42.431234, -8.643616], 5);
+    layerGroup = client.getLeafletLayerView();
+  });
+
+  afterEach(function () {
+    document.getElementById('map').remove();
+  });
+
+  describe('addTo', function () {
+    it('should add a leaflet layer to the map', function () {
+      expect(countLeafletLayers(map)).toEqual(0);
+
+      layerGroup.addTo(map);
+
+      expect(countLeafletLayers(map)).toEqual(1);
+    });
+  });
+
+  describe('removeFrom', function () {
+    it('should remove the leaflet layer from the map', function () {
+      expect(countLeafletLayers(map)).toEqual(0);
+
+      layerGroup.addTo(map);
+
+      expect(countLeafletLayers(map)).toEqual(1);
+
+      layerGroup.removeFrom(map);
+
+      expect(countLeafletLayers(map)).toEqual(0);
+    });
+  });
+
+  describe('layer events', function () {
```

Comment:
Maybe we could create once(using beforeAll) the `internalEvent` and `externalEvent` since they are shared in almost all tests